frontend/src/components/fillers/OrderDetailSection.tsx

```diff
@@ -40,19 +40,17 @@ export function OrderDetailSection({unselectOrder,onUpdate}: OrderDetailsProps)
 
     const changeState = (orderStatus: OrderStatus) => {
         if (orderStatus === OrderStatus.ACCEPTED
-            || orderStatus === OrderStatus.REJECTED
             || orderStatus === OrderStatus.CANCELLED
             || orderStatus === OrderStatus.NEEDS_INFO
             || orderStatus === OrderStatus.COMPLETED) {
             updateStateApi.request(selectedOrder!.uuid, orderStatus)
         }
-        if (orderStatus === OrderStatus.REJECTED
-            || orderStatus === OrderStatus.CANCELLED ) {
+        if (orderStatus === OrderStatus.CANCELLED ) {
             doForceRefresh();
             if (unselectOrder) {
                 unselectOrder();
             }
-            onUpdate();
+            updateStateApi.request(selectedOrder!.uuid, orderStatus)
```

Comment:
@milesressler , I'm not sure if this is the correct action to use as a replacement.